### ISSUE TITLE
External wireless on shieldtablet

### ISSUE THREAD

**Issue by hightechstl** [offensive-security/kali-nethunter#846]
### OS version (KitKat/Lollipop/Marshmallow/Nougat):
6.0.1 CM13

### Built from repo (date and build command) or downloaded from website (links):
kernel-nethunter-shieldtablet-marshmallow-3.15.4-20170211-0304.zip
From: 
https://build.nethunter.com/nightly/3.15.4-20170211-0304/

### Output of `cat /proc/version` in adb shell or NetHunter Terminal:

cat /proc/version
Linux version 3.10.67+ (buildbot@lusankya.tcpc.lan) (gcc version 4.8 (GCC) ) #1 SMP PREEMPT Fri Aug 19 21:37:02 CDT 2016

## Issue:
Unable to use external wireless. Always says device not found. lsusb shows the adapter though. Just can't get the interface to load....

 lsusb
Bus 002 Device 002: ID 148f:3070 Ralink Technology, Corp. RT2870/RT3070 Wireless Adapter
Bus 002 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 001 Device 001: ID 1d6b:0001 Linux Foundation 1.1 root hub

ifconfig wlan1 up
wlan1: ERROR while getting interface flags: No such device

ip link set wlan1
Cannot find device "wlan1"

dmesg can be found at 
https://pastebin.com/g3mTg3Vn

**Comment by hightechstl** [comment]
@binkybear hope you can shed some light in my mistakes

**Comment by simonpunk** [comment]
your os is CM, I think there is a chance that it could not be applied to your phone as the zip title didnt specify its I exclusive for CM

**Comment by hightechstl** [comment]
@simonpunk I've tried Stock 6.0.1 and cm13. Both have the same issue.

https://github.com/offensive-security/kali-nethunter/wiki

This page shows Supported devices list as shield tablet running 6.0.1 OR cm13

**Comment by simonpunk** [comment]
@hightechstl, so I think there are other possible cases
1. the nethunter kernel was not built from the source of your current kernel, that once happened to my nexus 6 until I flashed the latest offical factory image and compiled the latest kernel source, then it worked again for me.
2. wrong patch or configuration before compiling the kernel.
3. or the interface for your externel wireless adapter is wlan0?

I think the first case would likely happen, so you better flash the latest offical image first, and compile the nethunter kernel using the latest kernel source. I am not professional or expert tho, just some suggestions.

p.s. cm always appears to be buggy, and they didnt provide any updates and supports already. so I would suggest using original OS.

**Comment by hightechstl** [comment]
So I used the kernel listed here

https://build.nethunter.com/nightly/3.15.4-20170211-0304/

which I would imagine is the official image considering where I'm downloading it from....

I just flashed back to stock 6.0.1, flashed the kernel-nethunter-shieldtablet-marshmallow-3.15.4-20170211-0304.zip from the link above

Then flashed nethunter-generic-armhf-kalifs-full-rolling-3.15.4-20170211-0304.zip

Issue still remains. Says that wlan1 does not exist, but lsusb shows my external wireless adapter...

**Comment by simonpunk** [comment]
like I said, even you r running 6.0.1, doesnt mean there is only one build version of 6.0.1. for example, the nethunter kernel they built was from the source of build number 123456, but your factory kernel's build number is 456789, then there will be high possibility of bugs coming up.

**Comment by hightechstl** [comment]
So I re-flashed everything and my problem turned out to be order of operations.........

I set this all up following: https://forum.xda-developers.com/shield-tablet/development/kali-nethunter-nvidia-shield-tablet-t3356138


in their install steps they say:

2. Flash kernel-only/kernel-nethunter-shieldtablet and chroot-full/nethunter-generic-armhf-kalifs-full-rolling zips in any order.   <------ note the last 3 words.....


I then noticed on the nethunter wiki it says KERNEL MUST BE FLASHED LAST....

I wiped back to stock 6.0.1 on my nvidia shield, then flashed in this order:

1) nethunter-generic-armhf-kalifs-full-rolling-3.15.4-20170211-0304.zip
2) kernel-nethunter-shieldtablet-marshmallow-3.15.4-20170211-0304.zip
3) reboot and done


I hope this helps anyone else who has this same or similar issue!
